src/kvstore/kvstore_dist.h

```diff
@@ -467,6 +554,94 @@ class KVStoreDist : public KVStoreLocal {
     return pskv;
   }
 
+  /**
+   * \brief Convert to keys in ps for compressed values
+   * Divides original array into equal parts for each server
+   * Populates both push and pull pskv on first call
+   */
+  inline PSKV& EncodeCompressedKey(int key, size_t original_size, bool is_push) {
```

Comment:
Why does compressed value need different keys?

Reply:
Here, we compute sizes of values to send to server. They will be 1/16th for push. 
Sharding will also be slightly different because of differences in rounding up original_size/num_servers, and compressed_size/num_servers . 
Even the pull is affected because of the slight difference in sharding